backend/fetchArrivals.ts

```diff
@@ -0,0 +1,107 @@
+import axios from 'axios';
+export type ArrivalInfo = {
+    stationName: string;
+    lineName: string;
+    destinationName: string;
+    timeToStationMinutes: string;
+    towards: string;
+}
+
+type stopPoint = {
+    naptanId: string;
+    distance: number;
+}
+
+export async function getArrivalsGivenPostCode(postCode: string): Promise<ArrivalInfo[][] | string | null> {
+    const latLong = await getLatLongGivenPostCode(postCode);
+    if(latLong === null) return "Invalid Postcode";
+    const stopPoints = await getStopPointsGivenLatLong(latLong);
+    if(stopPoints === null) return "Bus Stops not found";
+
+    //Sort bus stops by nearest to farthest
+    stopPoints.sort((a, b) => a.distance - b.distance);
+
+    //Get the two nearest bus stops
+    const nearestTwoStops = stopPoints.slice(0,2);
+
+    //Get bus arrival information for each of the bus stops
+    const arrivalsPotential = await Promise.all(
+        nearestTwoStops.map(async (item) => {
+        const tempArrivals = await getArrivalsGivenStopPoint(item.naptanId);
+        return tempArrivals ?? []; // return empty array if null
+        })
+    );
+
+    //Checking for empty arrays in case of null return from API call
+    var arrivals: ArrivalInfo[][] = [];
+    for(const point of arrivalsPotential) {
+        if (point.length === 0) {
+            continue;
+        }
+        arrivals.push(point);
+    }
+    if (arrivals.length === 0) return "Invalid bus stops or no buses found for bus stops";
+    return arrivals;
+}
+
+async function getArrivalsGivenStopPoint(stopPoint: string): Promise<ArrivalInfo[] | null> {
+  try {
+    const response = await axios.get(`https://api.tfl.gov.uk/StopPoint/${stopPoint}/Arrivals?app_key=${api_key}`);
+
+    //Sort bus arrivals by earliest to latest
+    response.data.sort((a: { timeToStation: number; }, b: { timeToStation: number; }) => a.timeToStation - b.timeToStation);
+
+    //Get the first five buses at given stop
+    const firstFiveRows = response.data.slice(0,5);
+
+    //Return as ArrivalInfo type
+    return firstFiveRows.map((item: any): ArrivalInfo => {
+        var {stationName, lineName, destinationName, timeToStation, towards} = item;
+
+        //timeToStation gives a value whose unit is seconds, so convert this to a readable string of the form 'Xm Ys'
+        var timeToStationMinutes = `${Math.floor(timeToStation/60)}m ${timeToStation%60}s`;
+        return {stationName, lineName, destinationName, timeToStationMinutes, towards};
+    });

```

Comment:
this is quite a large map function. You could shrink it down with some different syntax
```suggestion
    return firstFiveRows.map((item: any): ArrivalInfo => (
    {
       stationName: item.stationName,
       lineName: item.LineName, // etc.
       timeToStationInSeconds: timeToStation, 
    }));

```

Reply:
Changed syntax as suggested.